src/mumble/MainWindow.cpp

```diff
@@ -1835,6 +1835,14 @@ void MainWindow::on_qaQuit_triggered() {
 	this->close();
 }
 
+void MainWindow::on_qaShow_triggered() {
+    if (isMinimized())
+        setWindowState((windowState() & ~Qt::WindowMinimized) | Qt::WindowActive);
+    show();
+    raise();
+    activateWindow();
+}
```

Comment:
https://github.com/mumble-voip/mumble/blob/ba087869df8416117584bac666a6dd619e2c838a/src/mumble/MainWindow.cpp#L3055-L3061

Since `on_Icon_messageClicked()` does the same thing as `on_qaShow_triggered()`, I would create a new function called `showWindow()` and then connect the two signals (`qaShow::triggered()` and `Icon::messageClicked()`) to that.

Reply:
The two slots (functions) are not necessary, because they just call `showRaiseWindow()`, but it's not a problem.